src/pkgcore/ebuild/ebd.py

```diff
@@ -679,7 +679,10 @@ def _setup_distfiles(self):
         # fetch distfiles
         if not self.verified_files:
             ops = self.domain.pkg_operations(self.pkg, observer=self.observer)
-            ops.fetch()
+            if ops.fetch():
+                # this break encapsulation and should be refactored.  Trace
+                # f35f2 and 6561eac for where this was refactored.
+                self.verified_files = ops.verified_files
```

Comment:
First, note that nothing in this method actually updates `self.verified_files` without my change.  That means the conditional that follows lien 688- will not fire unless something invoked by `ops.fetch()` mutates this instance.

Take a look at https://github.com/pkgcore/pkgcore/commit/f35f2c#diff-9e1b9f482d50cb7dabc3871cae3e57f9366c304bb8eaf9f0fae77a576ad91992L685-L690 ; you'll see the previous code broke encapsulation and transfered the `verified_files` mutable from the fetcher directly, and that that commit tried to feed it back through the operations API.

That broke things in a different way, which lead to https://github.com/pkgcore/pkgcore/commit/6561eacc4225f0b602d1612ee02b2b9d20d7a93d , which is where we're at now.

I honestly do not see anyway that this codepath could work and I suspect if you trace what I'm pointing, you'll reach a similar conclusion.

https://github.com/pkgcore/pkgcore/blob/master/src/pkgcore/ebuild/ebd.py#L670-L676 also shows the DISTDIR setup; it's basically the samething as the 'saviour' days, it uses a shadow $DISTDIR, and the host configuration can't really override that.

The only way this pathway would work is if `verified_files` was pre-populated and being passed in, or if the build init was doing something I'm not finding.  See https://github.com/pkgcore/pkgcore/blob/master/src/pkgcore/ebuild/ebd.py#L605-L608 for what I'm referring to.